### PR TITLE
feature(myclirc): use myclirc in local directory first if it exists

### PR DESCRIPTION
**Pull request by KeatingSmith**
## Description

resolve issue #648

Search .myclirc in same and directories above.

I request an ability from mycli tool to search it's .myclirc file in local directory, directory above till ${HOME} or /.

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by meeuw** [comment]
Nice feature! But I think it would be better to append `os.path.join(os.getcwd(), ".myclirc")` to `config_files` at https://github.com/dbcli/mycli/blob/1060644cc72ab1e0f2161044ff6c8d2ea5b83902/mycli/main.py#L111

**Comment by tsroten** [comment]
> 
> 
> Nice feature! But I think it would be better to append `os.path.join(os.getcwd(), ".myclirc")` to `config_files` at
> 
> [mycli/mycli/main.py](https://github.com/dbcli/mycli/blob/1060644cc72ab1e0f2161044ff6c8d2ea5b83902/mycli/main.py#L111)
> 
> Line 111 in [1060644](/dbcli/mycli/commit/1060644cc72ab1e0f2161044ff6c8d2ea5b83902)
>  config_files = ([self.default_config_file] + self.system_config_files +

Great idea @meeuw 😄. Plus, we need to make sure that people can still specify the myclirc file on the command-line via the `--myclirc` argument.


@KeatingSmith Thanks for working on this feature! This is a neat idea. 😃

**Comment by KeatingSmith** [comment]
I have change it, and test it.

File: mycli/main.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ class MyCli(object):
     ]
 
     default_config_file = os.path.join(PACKAGE_ROOT, 'myclirc')
-
+    pwd_config_file = os.path.join(os.getcwd(), "")
```
Comment (meeuw) [inline mycli/main.py]: I think you'll need to join the cwd and ".myclirc" instead of an empty string...

File: changelog.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Features:
 * Set `program_name` connection attribute (Thanks: [Dick Marinus]).
 * Use `return` to terminate a generator (Thanks: [Zhongyang Guan]).
 * Add `SAVEPOINT` to SQLCompleter (Thanks: [Huachao Mao]).
+* Seach `${PWD}/.myclirc`, then `${HOMW}/.myclirc`, last `/etc/.myclirc` (Thanks: [QiaoHao Peng])
```
Comment (meeuw) [inline changelog.md]: `HOMW` should be `HOME` ?

**Pull request merged**